src/lib/auth/AuthProviders/OAuthAuthenticator/index.ts

```diff
@@ -0,0 +1,122 @@
+import axios from 'axios'
+import { createHash } from 'crypto'
+import isWsl from 'is-wsl'
+import jwt from 'jsonwebtoken'
+import opn from 'opn'
+import { join } from 'path'
+import { logger } from '../../../../api'
+import { VTEXID } from '../../../../api/clients/IOClients/external/VTEXID'
+import { storeUrl } from '../../../../api/storeUrl'
+import { ColorifyConstants } from '../../../constants/Colors'
+import { randomCryptoString } from '../../../utils/randomCryptoString'
+import { spawnUnblockingChildProcess } from '../../../utils/spawnUnblockingChildProcess'
+import { AuthProviderBase } from '../AuthProviderBase'
+import { LoginServer } from './LoginServer'
+
+export class OAuthAuthenticator extends AuthProviderBase {
+  public static readonly AUTH_TYPE = 'oauth'
+  private static readonly SECRET_ALPHABET = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-._~'
+
+  private static ADMIN_LOGIN_URL_PATH = '/_v/segment/admin-login/v1/login'
+  private static FALLBACK_AUTH_SERVER_LOGIN_URL_PATH = '/_v/private/auth-server/v1/login'
+
+  public async login(account: string) {
+    const secret = randomCryptoString(128, OAuthAuthenticator.SECRET_ALPHABET)
+    const secretHash = this.hashSecret(secret)
+    const vtexId = VTEXID.createClient({ account })
+
+    const loginServer = await LoginServer.create({
+      account,
+      secret,
+    })
+
+    try {
+      const loginState = await vtexId.startToolbeltLogin({
+        account,
+        secretHash,
+        loopbackUrl: loginServer.loginCallbackUrl,
+      })
+
+      loginServer.setLoginState(loginState)
+
+      const url = await this.loginUrl(account, loginState)
+      opn(url, { wait: false })
+
+      if (isWsl) {
+        logger.warn(
+          "We noticed you're using WSL, in which case you may face login issues depending on your WSL version."
+        )
+
+        logger.warn(
+          `If you do, make sure your Windows is up to date and try again (${ColorifyConstants.URL_INTERACTIVE(
```

Comment:
I don't know if it was intentional to show the whole URL, but if not you could use this [function](https://github.com/vtex/toolbelt/blob/59ce5fe364b6734f6619301f4194abb0e08e8acd/src/lib/utils/Messages.ts#L5) to show hyperlink

Reply:
I couldn't think of a text to use as the URL :( - and there's an issue with WSL -> as of now it doesnt support ansi url codes (I tested on WSL1 and WSL2, Ubuntu 20.04 and 18.04)

However if you have a suggestion to use as a text I would very much accept it :)

Reply:
This function previously checks the hyperlink support, would it be enough to solve this problem?

Well, I can only think in something simple like `here`, maybe @VictoriaMolgado can think about a better option

Reply:
The only problem here is that we know in this case the function output would be like the following, because of WSL:
```
If you do, make sure your Windows is up to date and try again. For update instructions check here (URL).
```
In this case, what you would prefer?
```
If you do, make sure your Windows is up to date and try again. For update instructions check here (URL).
```
or
```
If you do, make sure your Windows is up to date and try again. For update instructions check URL.
```

Reply:
We are inside the `if wsl`, for a moment I forgot about it 😆  Indeed, it looks better this way